diff -Nur perftest-4.2-old/src/raw_ethernet_resources.c perftest-4.2/src/raw_ethernet_resources.c
--- perftest-4.2-old/src/raw_ethernet_resources.c	2018-05-06 19:58:57.000000000 +0800
+++ perftest-4.2/src/raw_ethernet_resources.c	2021-08-03 15:00:40.000000000 +0800
@@ -54,7 +54,7 @@
 #include "raw_ethernet_resources.h"
 #include "config.h"
 
-struct perftest_parameters* duration_param;
+extern struct perftest_parameters* duration_param;
 
 int check_flow_steering_support(char *dev_name)
 {
